ERGA post-assembly QC training (#3908)
galaxyproject/training-material: topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.md

3. Genome size estimation: Accurate estimation of genome size is important for downstream analyses such as gene annotation and comparative genomics.
4. Contamination rate: Contamination occurs when foreign DNA sequences are introduced into the assembly, which can skew downstream analyses.

These metrics can help researchers to evaluate the quality of their eukaryotic genome assemblies and identify potential issues that may impact downstream analyses.
-->

In this tutorial you will learn how to implement the ERGA GPAQC pipeline, and how to interpretate the potential outcomes.

> <agenda-title></agenda-title>
>
> In this tutorial, we will cover:
>
> 1. TOC
> {:toc}
>
{: .agenda}

# Study cases

In this tutorial we will evaluate three genome assemblies, belonging to three different taxonomic groups, in order to illustrate the different scenarios that can be identified. The characteristics of each of them are described briefly below.

> ##### Case 1: ***Chondrosia reniformis***:

Reviewer: ```suggestion
<strong>Case 1: <em>Chondrosia reniformis</em></strong>
```

*Chondrosia reniformis* is a slow-growing marine sponge cosmopolitan species which can be found in the Mediterranean Sea and the eastern Atlantic Ocean in shallow waters; it is considered to playing an important ecological role in the marine ecosystem by filtering large volumes of water and providing habitat for other species ({% cite Voultsiadou2005 %}). The members of this species are gonochoristic and oviparous, whose physiology and behavious seems to be highly influence for the presence of endosymbiosis heterotrophic bacteria ({% cite sara1997 %}). 

![Figure 1: Distribution of data](../../images/post-assembly-QC/chondrosia.jpg "Histories corresponding to the three cases of study.")

The assembly is based on 70x PacBio HiFi and Arima2 Hi-C data generated by the Aquatic [Symbiosis Genomics Project](https://www.aquaticsymbiosisgenomics.org/). The assembly process included the following sequence of steps: initial PacBio assembly generation with Hifiasm, retained haplotig separation with purge_dups, and Hi-C based scaffolding with YaHS. The mitochondrial genome was assembled using MitoHiFi. Finally, the primary assembly was analysed and manually improved using gEVAL.

> ##### Case 2: ***Erythrolamprus reginae***:

Reviewer: ```suggestion
<strong>Case 2: <em>Erythrolamprus reginae</em></strong>
```

*Erythrolamprus reginae* is a species of colubrid snake found in South America. This species has been reported to include triploid individuals with parthenogenic reproduction; this type of seems to be associated with higher mutation rate tandem dupliation ({% cite Bogart1980 %}). 

![Figure 2: Distribution of data](../../images/post-assembly-QC/erytrocampus.jpg "Histories corresponding to the three cases of study.")

> The assembly used in this tutorial correspond to the curated primary assembly generated by the [VGP project](https://vertebrategenomesproject.org/), based on 34x Pacbio HiFi and Arima2 Hi-C data, by using the [VGP assembly pipeline](https://training.galaxyproject.org/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html). 

Reviewer: ```suggestion
The assembly used in this tutorial correspond to the curated primary assembly generated by the [VGP project](https://vertebrategenomesproject.org/), based on 34x Pacbio HiFi and Arima2 Hi-C data, by using the [VGP assembly pipeline]({% link topics/assembly/tutorials/vgp_genome_assembly/tutorial.html %}). 
```

> ##### Case 3: ***Eschrichtius robustus***:

Reviewer: ```suggestion
<strong>Case 3: <em>Eschrichtius robustus</em></strong>
```

*Eschrichtius robustus*, commonly known as the gray whale, is a species of whale found primarily in the North Pacific Ocean. Adult gray whales can reach lengths of up to 14.9 meters and weights of up to 36,000 kilograms. It is a diploid specie, genetically characterized by high homozygosity as a result of inbreeding ({% cite BrnicheOlsen2018 %}).

![Figure 3: Distribution of data](../../images/post-assembly-QC/eschrichtius.jpg "Histories corresponding to the three cases of study.")

> The assembly used in this tutorial correspond to the curated primary assembly generated by the [VGP project](https://vertebrategenomesproject.org/), based on 29x Pacbio HiFi data, by using the [VGP assembly pipeline](https://training.galaxyproject.org/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html). 

Reviewer: ```suggestion
The assembly used in this tutorial correspond to the curated primary assembly generated by the [VGP project](https://vertebrategenomesproject.org/), based on 29x Pacbio HiFi data, by using the [VGP assembly pipeline]({% link topics/assembly/tutorials/vgp_genome_assembly/tutorial.html %}). 
```

>       ```
>
>    - From **Rules** menu select `Add / Modify Column Definitions`
>       - Click `Add Definition` button and select `Name`: column `A`
>       - Click `Add Definition` button and select `URL`: column `B`
>       - Click `Add Definition` button and select `Type`: column `C`
>       - Click `Add Definition` button and select `Name Tag`: column `D`
>    - Click `Apply` and press <kbd>Upload</kbd>
>
{: .hands_on}

Once we have imported all the datasets, we will move each one to its correspondent history. 

> <hands-on-title>Upload data</hands-on-title>
>
> 1. Create three new empty histories, one for each specie.
> 2. Rename the histories as  `Case 1: *Chondrosia reniformis*`, `Case 2: *Erythrolamprus reginae*` and  `Case 3: *Eschrichtius robustus*`.
> 3. Click in **History options** and select `Show Histories Side-by-Side`
> 4. Click in `Select histories`, and include the histories corresponding to the three species.
> 5. Move the datasets to its correspondent history.
>
>    > <comment-title>Non-unique datasets</comment-title>
>    > Both the **Taxonomic_data** and the **Diamond_database** should be included in all of them.

> >    {: .details}

Reviewer: ```suggestion
>    {: .comment}
```

> <question-title>Hi-C contact map question</question-title>
>
> What can you say about the Hi-C map corresponding to *Chondrosia reniformis*?
>
> > <solution-title></solution-title>
> >
> > ![Figure 17: Contact Map Sponge](../../images/post-assembly-QC/pretext_sponge.png "Hi-C contact map of Chondrosia reniformis generated by Pretext.")
> >
> > The contact map of the sponge (*Chondrosia reniformis*) does contain widespread interactions all over the genome. Additionally a diagonal pattern can be observed. The widespread interactions all over the genome can be the result of experimental and technical issues or it can have a biological and funcitonal significance. 
> {: .solution}
>
{: .question}

# Assembly graph

Bandage is a tool to visualise de novo assembly graphs with connections. ({% cite Wick2015 %})

> <hands-on-title> Generate assembly graph </hands-on-title>
>

> 1. {% tool [gfastats](toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0) %} with the following parameters:

Reviewer: ```suggestion
> 1. {% tool [gfastats](toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0) %} with the following parameters:
```
